lib/mayaUsd/fileio/translators/translatorBlendShape.cpp

```diff
@@ -0,0 +1,299 @@
+//
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+#include "translatorBlendShape.h"
+
+#include <pxr/base/tf/diagnostic.h>
+#include <pxr/base/vt/types.h>
+#include <pxr/usd/usd/common.h>
+#include <pxr/usd/usdSkel/animQuery.h>
+#include <pxr/usd/usdSkel/animation.h>
+#include <pxr/usd/usdSkel/binding.h>
+#include <pxr/usd/usdSkel/bindingAPI.h>
+#include <pxr/usd/usdSkel/blendShape.h>
+#include <pxr/usd/usdSkel/root.h>
+
+#include <maya/MFnAnimCurve.h>
+#include <maya/MFnBlendShapeDeformer.h>
+#include <maya/MFnGeometryFilter.h>
+#include <maya/MItDependencyGraph.h>
+#include <maya/MPointArray.h>
+#include <maya/MStatus.h>
+
+#include <iterator>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+
+void _AddBlendShape(
+    MFnBlendShapeDeformer& blendShapeFn,
+    const MObject&         originalShape,
+    const MObject&         parentTransform,
+    const MPointArray&     originalPoints,
+    const VtIntArray&      pointIndices,
+    const std::string      name,
+    const VtVec3fArray     offsetArray,
+    int                    blendShapeTargetIndex,
+    float                  weight)
+{
+    MFnMesh ibMeshFn;
+    MStatus status;
+    auto    newShape = ibMeshFn.copy(originalShape, parentTransform, &status);
+
+    MPointArray ibPoints(originalPoints);
+    for (size_t pidx = 0; pidx < pointIndices.size(); ++pidx) {
+        int     index = pointIndices[pidx];
+        MPoint  original = originalPoints[index];
+        GfVec3f offset = offsetArray[pidx];
+
+        ibPoints.set(
+            MPoint(original.x + offset[0], original.y + offset[1], original.z + offset[2]), index);
+    }
+    ibMeshFn.setPoints(ibPoints);
+
+    MFnDependencyNode ibDepNodeFn;
+    ibDepNodeFn.setObject(newShape);
+    ibDepNodeFn.setName(MString(name.c_str()));
+
+    blendShapeFn.addTarget(originalShape, blendShapeTargetIndex, newShape, weight);
+    ibMeshFn.setIntermediateObject(true);
```

Comment:
I suppose the order of setting it as intermediate after instead of before adding it as a target does not matter.

Reply:
In this case, what really matter is how we setup the plugs connection. The order in which the nodes are created aren't essential.